### ISSUE TITLE
Docker Swarm Tutorial - Script Setup Failure on Windows

### ISSUE THREAD

**Issue by wisehackermonkey** [docker/labs#308]
# Docker Swarm Tutorial
[Image of Install Here ](http://imgur.com/a/kbgnD)

when i download the swarm files and run the setup script for powershell (admin) it throws errors
and does not install .

So i tried  installing the image what was used 
>docker pull virtualbox
"image not found" or somthing like it

what i did
1.powershell as admin
>git clone https://github.com/docker/labs.git
>cd \labs\swarm-mode\beginner-tutorial
>./swarm-node-hyperv-setup.ps1

#### Docker version 17.06.0-ce, build 02c1d87

Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 11
Server Version: 17.06.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 17
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: r3csy4gabmkraq1f3ei7b1o6w
 Is Manager: true
 ClusterID: 86s6elax4gbfvuc9z9d4f2z6t
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Root Rotation In Progress: false
 Node Address: 192.168.65.2
 Manager Addresses:
  192.168.65.2:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.36-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.934GiB
Name: moby
ID: BFKK:PVAY:QVC3:PKHS:L7D3:WDA6:V6GX:LSG3:4BKC:GUJT:C5QA:3SZ7
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 30
 Goroutines: 137
 System Time: 2017-07-16T07:41:48.9918852Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Comment by ManoMarks** [comment]
We could do a better job of explaining the Windows portion. And I'll work on that.

My first suggestion is to try this over on [Play with Docker](http://training.play-with-docker.com/swarm-mode-intro/) instead of trying it locally. Setting up multiple hyper-v virtual machines locally is never a scenario you're likely to do in the real world.

If you really want to go down this route, you'll see in the lab we linked off to some requirements:
[Setting up Microsoft Hyper-V Driver](https://docs.docker.com/machine/drivers/hyper-v/)
That will also lead you to making sure you have hyper-v installed on your system, which requires installing a switch. Here's the [Microsoft directions for doing this](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v). In the powershell script we provided, we called this "New Virtual Switch" but you'll have to replace that with whatever name you gave the switch when you set-up hyper-v.
